src/goto-programs/skolemization.cpp

```diff
@@ -0,0 +1,69 @@
+/// \file skolemization.h
+/// Rename variables in existentially quantified statements.
+/// Author: Diffblue Ltd.
+
+#include "skolemization.h"
+
+#include "goto_model.h"
+
+#include <ansi-c/expr2c.h>
+
+#include <util/expr_util.h>
+#include <util/fresh_symbol.h>
+#include <util/irep.h> // for id2string
+#include <util/mathematical_expr.h>
+
+void rename_variables_in_existential_quantifiers(goto_modelt &goto_model)
+{
+  for(auto &function : goto_model.goto_functions.function_map)
+  {
+    for(auto &instruction : function.second.body.instructions)
+    {
+      if(instruction.has_condition())
+      {
+        auto &cond = instruction.condition_nonconst();
+        recurse_on_condition(cond, goto_model);
+      }
+    }
+  }
+}
+
+void recurse_on_condition(exprt &cond, goto_modelt &model)
```

Comment:
:pick: `recurse_on_` just tells the reader how this function is implemented, not what it does with/to conditions.

Reply:
Ok, this is now resolved.